kernel/src/udp_port_table.rs

```diff
@@ -0,0 +1,249 @@
+//! UDP port table implementation for enforcing port binding. This table is
+//! checked when packets are sent/received by a UDPSendStruct/UDPReceiver.
+//! UdpPortBinding provides an opaque descriptor object that allows the holder
+//! interact with the bound port table. Only the holder of the UdpPortBinding
+//! object can interact with its own corresponding location in the bound port
+//! table. In order to bind to a particular port as sending/receiving, one must
+//! obtain the corresponding sender/receiving binding from UdpPortBinding.
```

Comment:
I'm not sure I understand what exactly the code in this file has to do. I think a higher-level description would be useful, as well as a description of what is stored here vs. what is stored elsewhere in a capsule.